src/chapters/enums/intro.rst

```diff
@@ -0,0 +1,346 @@
+
+   .. sourcecode:: java
+      :linenos:
+
+      public enum LogLevel {
+      
+         DEBUG,
+         INFO,
+         WARNING,
+         ERROR
+
+      }
+
+   An application can change the way it logs messages by changing the log level.
+
+.. admonition:: Example
+
+   An enum that you have :ref:`already encountered <request-method-example>` is ``RequestMapping``, which we have used to specify which HTTP methods a controller method can respond to. This enum has values ``RequestMapping.GET``, ``RequestMapping.POST``, ``RequestMapping.DELETE``, and so on.
+
+Adding Properties to Enums
+--------------------------
+
+It can sometimes be useful to add properties and methods to enum types, in order to associate data and behaviors with each value.  
+
+Let's consider the example of our ``Day`` enum from above. We can associate a user-friendly display name (such as ``"Saturday"`` for ``Day.SATURDAY``), along with a boolean representing its status as a weekday, just as you would in any other class.
+
+Let's include the relevant fields, getters, and a constructor. Note that the fields will be ``final`` since they should never change value. This also means that they don't need setters.
+
+.. sourcecode:: java
+   :linenos:
+
+   private final String displayName;
+   private final boolean weekday;
+
+   public Day (String displayName, boolean weekday) {
+      this.displayName = displayName;
+      this.weekday = weekday;
+   }
+
+   public String getDisplayName() {
+      return this.displayName;
+   }
+
+   public boolean isWeekday() {
+      return this.weekday;
+   }
+
+To specify the values of ``displayName`` and ``weekday`` for each enum value, we call the constructor when listing the values. The full class then looks like this:
+
+.. sourcecode:: java
+   :linenos:
+
+   public enum Day {
+
+      SUNDAY("Sunday", true),
+      MONDAY("Monday", false),
+      TUESDAY("Tuesday", false),
+      WEDNESDAY("Wednesday", false),
+      THURSDAY("Thursday", false),
+      FRIDAY("Friday", false),
+      SATURDAY("Saturday", true);
+
+      private final String displayName;
+      private final boolean weekday;
+
+      public Day (String displayName, boolean weekday) {
+         this.displayName = displayName;
+         this.weekday = weekday;
+      }
+
+      public String getDisplayName() {
+         return this.displayName;
+      }
+
+      public boolean isWeekday() {
+         return this.weekday;
+      }
+
+   }
+
+.. admonition:: Note
+
+   Notice that we now have to add a semi-colon after our list of enum values.
+
+
+Check Your Understanding
+------------------------
+
+.. admonition:: Question
+
+   We mentioned above that all classes define a data type. Is the inverse of this statement true? In other words, do all data types correspond to a class? (*Hint:* Try to think of a data type that is NOT a class.)
```

Comment:
possible answer choices:
#. Yes, everything in Java is a class.
#. No, primitive data types are not classes.
.. ans: b, primitive data types are not classes.